engine/overworld/weather.asm

```diff
@@ -0,0 +1,778 @@
+DoOverworldWeather:
+	push de
+	push hl
+	push bc
+	ld hl, wWeatherFlags
+	bit OW_WEATHER_DISABLED_F, [hl]
+	jr nz, .done
+	ld a, [wOverworldWeatherDelay]
+	and %1 ; 30 fps
```

Comment:
What is the overall behavior of `wOverworldWeatherDelay`? It steadily increments... up to some maximum? Does it wrap or cap?

It's not immediately obvious what the connection is between the low bit and "30 fps"; IMO "`; every other/odd frame (30 fps)`" would be an improvement. (Like for people who forgot the game is 60 FPS now. :P )

Reply:
Renamed to `wOverworldWeatherTimer`; it's also currently used to time cleaning up rain splashes.

```
	; rolling counter from 0 to 255 used to run weather
	; in 30fps or every odd frame.
	ld a, [wOverworldWeatherTimer]
	and %1
	jr z, .done
```